packages/strapi-plugin-graphql/services/Loaders.js

```diff
@@ -0,0 +1,187 @@
+          reject(e);
+        }
+      });
+    }, {
+      cacheKeyFn: (key) => {
+        return _.isObjectLike(key) ? JSON.stringify(_.cloneDeep(key)) : key;
+      }
+    });
+  },
+
+  mapData: function(model, originalMap, map, results) {
+    // Use map to re-dispatch data correctly based on initial keys.
+    return originalMap.map((query, index) => {
+      // Find the index of where we should extract the results.
+      const indexResults = map.findIndex(queryMap => queryMap.indexOf(index) !== -1);
+      const data = results[indexResults];
+
+      // Retrieving referring model.
+      const ref = this.retrieveModel(model, query.options.source);
+
+      if (query.single) {
+        // Return object instead of array for one-to-many relationship.
+        return data.find(entry => entry[ref.primaryKey].toString() === (query.params[ref.primaryKey] || '').toString());
+      }
+
+      // Extracting ids from original request to map with query results.
+      const ids = (() => {
+        if ( _.get(query.options, `query.${ref.primaryKey}`)) {
+          return  _.get(query.options, `query.${ref.primaryKey}`);
+        }
+
+        // Single object to retrieve (one-to-many).
+        const alias = _.first(Object.keys(query.options.query));
+
+        return {
+          alias,
+          value: _.get(query.options, `query.${alias}`)
+        };
+      })();
+      
+      if (!_.isArray(ids)) {
+        return data
+          .filter(entry => entry[ids.alias].toString() === ids.value.toString())
+          .slice((query.options.start || query.options.skip), (query.options.start || query.options.skip) + query.options.limit);
+      }
+
+      // Critical: don't touch this part until you truly understand what you're doing.
+      // The data array takes care of the sorting of the entries. It explains why we are looping from this array and not the `ids` array.
+      // Then, we're applying the `limit`, `start` and `skip` argument.
+      return data
+        .filter(entry => entry !== undefined)
+        .filter(entry => ids.map(id => id.toString()).includes(entry[ref.primaryKey].toString()))
+        .slice((query.options.start || query.options.skip), (query.options.start || query.options.skip) + query.options.limit);
+    });
+  },
+
+  makeQuery: async function(model, query = {}) {
+    if (_.isEmpty(query.ids)) {
+      return [];
+    }
+
+    const ref = this.retrieveModel(model, query.options.source);
+
+    // Construct parameters object sent to the Content Manager service.
+    // We are faking the `start`, `skip` and `limit` argument because it doesn't make sense because we are merging different requests in one.
+    // Note: we're trying to avoid useless populate for performances. Please be careful if you're updating this part.
+    const params = {
+      ...query.options,
+      populate: ref.associations.filter(a => !a.dominant && _.isEmpty(a.model)).map(a => a.alias),
+      query: {},
+      start: 0,
+      skip: 0,
```

Comment:
Why both (skip and start) aren't they the same?

Reply:
No idea... I wasn't the job to rework the filters with this PR but we could definitely make things clearer and this part. I was confusing to me as well.